sycl/doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc

```diff
@@ -219,66 +220,309 @@ q.parallel_for(nd_range<2>(G, L), [=](nd_item<2> item)
+    uint32_t max_nsize;
+    uint32_t max_ksize;
+    matrix_type atype;
+    matrix_type btype;
+    matrix_type ctype;
+    uint32_t msize;
+    uint32_t nsize;
+    uint32_t ksize;
+  };
+  static constexpr combination combinations[] = {
+      {16, 16, (sizeof(Ta) == 1) ? 64 : 32}};
+  static constexpr int num_combinations =
+      sizeof(combinations) / sizeof(combination);
+};
+
+// General query:
+// types are not given, no default sizes and no implicit matrix construction
+template <int M, int N, int K>
+struct tpu_params<tpu::amx, void, void, void, M, N, K> {
+  static constexpr std::size_t defaultM = -1; // depends on the type
+  static constexpr std::size_t defaultN = -1;
+  static constexpr std::size_t defaultK = -1;
+
+  bool dynamic_p = false; // should be true in future implementations because
+                          // AMX hardware supports dynamic sizes
+  uint32_t numtiles = 8;
+  scope_t scope = scope_t::sub_group;
+  struct combination {
+    uint32_t max_msize;
+    uint32_t max_nsize;
+    uint32_t max_ksize;
+    matrix_type atype;
+    matrix_type btype;
+    matrix_type ctype;
+    uint32_t msize;
+    uint32_t nsize;
+    uint32_t ksize;
+  };
+  using mt = matrix_type;
+  static constexpr combination combinations[] = {
+      {16, 16, 64, mt::sint8, mt::sint8, mt::sint32},
+      {16, 16, 64, mt::sint8, mt::uint8, mt::sint32},
+      {16, 16, 64, mt::uint8, mt::sint8, mt::sint32},
+      {16, 16, 64, mt::uint8, mt::uint8, mt::sint32},
+      {16, 16, 32, mt::bf16, mt::bf16, mt::fp32}};
+  static constexpr int num_combinations =
+      sizeof(combinations) / sizeof(combination);
+};
+
+
+enum class tpu {
+  dpas,
+  amx
+};
 
-### Current Implementation Restrictions
-This section summarizes the specific features that this implementation supports. In future versions of this API and implementation, the expectation is to provide a query interface to guide the usage of this API. 
+enum class matrix_type {
+bf16,
+fp16,
+fp19,  // tfloat32
+fp32,
+fp64,
+sint2,
+sint4,
+sint8,
+sint16,
+sint32,
+sint64,
+uint2,
+uint4,
+uint8,
+uint16,
+uint32,
+uint64
+};
 
-#### Type, Sizes, and Layouts
-The types supported by this AMX implementation are restricted to the types that AMX hardware support. Although the AMX hardware supports 2d tiles with a maximum size of 16x64 bytes, this current implementation can handle any size. If the matrix size is bigger than 1024 bytes, it will be stored in memory rather than mapped to a 2d tile. Performance penalty may occur in this case. In order to get the best performance with this implementation, matrix sizes should be no larger than 16x64 bytes and B matrix should be already packed (put in VNNI format).
+enum class scope_t {
+sub_group,
+work_group
+};
+}
+```
 
-More specifically, the following operation C = A*B+C can be performed on AMX with this interface where:
 
-A(int8, any-size, row_major), B(int8, any-size, packed_b), C(int32, any-size, row_major)
+=== Valid or not Example:
```

Comment:
I found this subsection title a little confusing.  I'm nit-picking, but I don't see any example of querying here, and it's not immediately obvious from this example where the assertion happens.